### PR TITLE
DAOS-14052 test: Update cart_logtest usage in launch.py

### PR DESCRIPTION
**Pull request by ashleypittman**
Update log_test to not check for descriptor consistency when parsing
ftest logs.  As ftest debug masks change on live systems these tests
aren't valid and will fail if RPCs are in flight during the change.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
@brianjmurrell or @phender 
When this fails I can see the stdout from cart_logtest.py as in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13005/5/testReport/FTEST_launch/datamover-negative/Test___Functional_on_EL_8___01___datamover_negative_py/ but when there's no failure then I can't see the output of this command anywhere.

This PR is a bug-fix but also removes some technical debt and improves the layout of the code as well, there were some further potential improvements I identified in the ticket but I'll handle those separately.

File: src/tests/ftest/cart/util/cart_logtest.py [hunk 1]
```diff
@@ -565,29 +527,9 @@ def _check_pid_from_log_file(self, pid, abort_on_warning, leak_wf, show_memleaks
...
-                    (pointer, _) = line.realloc_pointers()
-                if pointer in active_desc:
-                    if show_line(line, 'NORMAL', 'descriptor not freed', custom=leak_wf):
-                        print(f'Memory address is {pointer}')
-                    del active_desc[pointer]
-                else:
-                    if show_line(line, 'NORMAL', 'memory not freed', custom=leak_wf):
-                        print(f'Memory address is {pointer}')
-                lost_memory = True
+                    fi_count, (fi_count / mem_r.memsize.count) * 100))
```
[the remark this answers is not in the document]
Comment (ashleypittman) [reply src/tests/ftest/cart/util/cart_logtest.py:532]: The same for two lines up.  I guess fi_count is only non-zero for the FI testing, and that isn't run under ftest.

I considered making a dummp MemReporting class that would look like MemReporting but not do anything, but that seemed like a hack.  There's some technical debt in this file though that this PR only partly addresses (and highlights some other issues).  I think probably the descriptor tracing should be in it's own class as well but I viewed that as out-of-scope, and then MemReporting if enabled would take that as a input and share some of the same data structures.  I'll work on resolving that next but I think this PR stands as-is and this mem_r use is safe as it's guarded by the fi_count test above.

**Review by daltonbohning — APPROVED** [review]

**Comment by ashleypittman** [comment]
@phender any comments on this?

**Comment by ashleypittman** [comment]
Can I get some more reviews on this please?

**Pull request merged**